src/OSPSuite.Core/Domain/Module.cs

```diff
@@ -16,11 +17,27 @@ public class Module : ObjectBase, IEnumerable<IBuildingBlock>
 
       private T buildingBlockByType<T>() where T : IBuildingBlock => _buildingBlocks.OfType<T>().SingleOrDefault();
       private IReadOnlyList<T> buildingBlocksByType<T>() where T : IBuildingBlock => _buildingBlocks.OfType<T>().ToList();
+      private readonly ExtendedPropertyStore<Module> _extendedPropertyStore;
 
+      /// <summary>
+      /// Module is a PKSim module if created in PKSim and the module version
+      /// matches the version when it was first imported
+      /// </summary>
       public bool IsPKSimModule =>
-         ExtendedProperties.Contains(Constants.PK_SIM_VERSION) &&
-         ExtendedProperties.Contains(Constants.PK_SIM_MODULE_IMPORT_VERSION) &&
-         ExtendedProperties[Constants.PK_SIM_MODULE_IMPORT_VERSION].ValueAsObject.Equals(Version);
+         !string.IsNullOrEmpty(PKSimVersion) && 
+         Equals(Version, _extendedPropertyStore.Get(x => x.ModuleImportVersion));
+
+      public string ModuleImportVersion
+      {
+         get { return _extendedPropertyStore.Get(x => x.ModuleImportVersion); }
```

Comment:
Probably can be on it one line now using new syntax


I forgot something. Why are those extended properties of the module and not just properties? Extended properties are more dynamic stuff that you don't really control. In this case...seems like we know about then after all? Mostly a question for me